angular-register.ts

```diff
@@ -0,0 +1,30 @@
+declare interface NoopDecorator {
+  <T>(type: T): T;
+  (): (...args: any[]) => void;
```

Comment:
I get that this is a more "catch-all" behavior. But according to the TypeScript doc, a strict typing of this function would be:

```ts
(): (_target: any, _propertyKey: string, _descriptor?: PropertyDescriptor) => void;
```

Why did you chose the "any function" type instead?

Reply:
Given that the operator doesn't do anything this was just to reduce the mental overhead of unused values.

We could use the proper typings here, but would it result any any differences given that the values are never used?

Reply:
Using the full prototype is a way to tell TypeScript what is allowed to be decorated. This is admittedly a very narrow edge case, but if someone accidentally uses the wrong decorator here:

```ts
@Override()
function someRandomFunction() {

}
```

TS knows the decorator is not supposed to be applied to a function and will emit some kind of error.

With a catch-all type, the decorator can be placed anywhere and TS will do nothing.

Reply:
Updated to include proper strict typing.